e2e-tests/dcp-issuance-tests/src/test/java/org/eclipse/edc/identityhub/tests/dcp/flow/DcpAnonymousIssuanceFlowEndToEndTest.java

```diff
@@ -140,10 +146,26 @@ void issuanceFlow(IssuerService issuer, IdentityHub identityHub) {
                     .atMost(TIMEOUT)
                     .untilAsserted(() -> assertThat(identityHub.getCredentialRequestForParticipant(PARTICIPANT_ID)).hasSize(1)
                             .allSatisfy(t -> {
-                                assertThat(t.getState()).isEqualTo(HolderRequestState.ISSUED.code());
+                                assertThat(t.getState()).isEqualTo(HolderRequestState.REQUESTED.code());
                                 assertThat(t.getHolderPid()).isEqualTo("test-request-id");
                             }));
 
+            // wait for the issuance process to be pending on the issuer side
```

Comment:
please move the pending guard test into a second test method `issuanceFlow_withGuard` leaving the original method untouched. Yes, that might duplicate some code, but that second method would only need to assert that the issuance process does not advance automatically and remains in the given state.